### PR TITLE
fetching the associated PR instead of the latest for publishing changes to changelog

### LINKED ISSUE #715: Checking the triggering PR instead of the latest when merging PRs with changelog
Original problem here: 
 https://github.com/ospin-web-dev/hambda/issues/321

Fix can be applied by not fetching the latest PR to check for changelog, but the one that actually triggered the pipeline. This can be solved using the `CIRCLE_PULL_REQUEST` variable: 

https://circleci.com/docs/2.0/env-vars/?utm_medium=SEM&utm_source=gnb&utm_campaign=SEM-gb-DSA-Eng-ni&utm_content=&utm_term=dynamicSearch-&gclid=EAIaIQobChMI-u_2hdqI6QIVjbd3Ch0sagUzEAAYASAAEgKpb_D_BwE#built-in-environment-variables

### PR DESCRIPTION
**Pull request by vdiezel**
closes #715 

For the publish of changes - which does nothing but making the icon for changelog in the sidebar red when a PR with the `changelog` label gets merged - we now check the PR that actually triggered the CI/CD suite instead of fetching the latest PR, because the old approach causes issues when multiple PRs are in the CI/CD pipeline leading to changelog sometimes not to be published (see ticket). 

Circle CI provides a built in environmental variables with the PR URI, so I just parse the PR number out there and call the GitHub API with it.

### CODE REVIEW

File: .circleci/publishChanges.js [hunk 1]
```diff
@@ -1,45 +1,46 @@
 const fetch = require('node-fetch')
 const AWS = require('aws-sdk')
 
-const CLOSED_PRS_ENDPOINT = `https://api.github.com/repos/ospin-web-dev/REACTor/pulls?state=closed`
```
Comment (DanielSeehausen) [inline .circleci/publishChanges.js:4]: just curious why we no longer use closed

Comment (vdiezel) [reply .circleci/publishChanges.js:4]: It fetches the PR that triggered the pipeline, using `state=closed` is not required as information here anymore, as a single PR is fetched by its Id (aka number)

Comment (DanielSeehausen) [reply .circleci/publishChanges.js:4]: noice

File: .circleci/publishChanges.js [hunk 2]
```diff
@@ -1,45 +1,46 @@
...
+const PRS_ENDPOINT = `https://api.github.com/repos/ospin-web-dev/REACTor/pulls/`
 
 // ENV VARS passed in from the bash via process.argv
 // they are stored in the circleCI Project COnfiguration of REACTor
```
Comment (ConorTighe1995) [inline .circleci/publishChanges.js:7]: Small typo here on `Configuration`

no biggie tho

File: .circleci/publishChanges.js [hunk 3]
```diff
@@ -1,45 +1,46 @@
...
+
+const PR_PATH_SEGMENTS = CIRCLE_PULL_REQUEST.split('/')
+const PR_NUMBER = PR_PATH_SEGMENTS.pop()
 
 AWS.config = { region: OSPIN_AWS_REGION }
 
 const getFunctionName = () => {
-  if (DEPLOYMENT_ENV !== 'dev' && DEPLOYMENT_ENV !== 'prod'){
+  if (DEPLOYMENT_ENV !== 'dev' && DEPLOYMENT_ENV !== 'prod') {
     throw new Error(`Unknown deployment environment provided.`)
```
Comment (ConorTighe1995) [inline .circleci/publishChanges.js:21]: If we throw and error here will this have any effect on the CI-CD process? will it stop things or just not publish the change-log?

Comment (vdiezel) [reply .circleci/publishChanges.js:21]: It will only fail the publish step

File: .circleci/publishChanges.js (hunk 2 above)
Comment (vdiezel) [reply .circleci/publishChanges.js:7]: thanks!

**Pull request merged**